apps/etterlatte-beregning/src/main/kotlin/beregning/BeregningService.kt

```diff
@@ -59,135 +57,210 @@ class BeregningService(
-                            grunnbelopMnd = Grunnbeloep.hentGjeldendeG(datoFom).grunnbeløpPerMåned,
-                            grunnbelop = Grunnbeloep.hentGjeldendeG(datoFom).grunnbeløp,
-                            trygdetid = 40 // TODO: Må fikses før vi tar imot saker som IKKE har 40 års trygdetid
+                            soeskenFlokk = emptyList(),
+                            grunnbelopMnd = grunnbeloep.grunnbeløpPerMåned,
+                            grunnbelop = grunnbeloep.grunnbeløp,
+                            trygdetid = barnepensjonGrunnlag.avdoedForelder.verdi.trygdetid.toInt()
                         )
                     ),
-                    beregnetDato = LocalDateTime.now().toTidspunkt(norskTidssone),
+                    beregnetDato = Tidspunkt.now(),
                     grunnlagMetadata = grunnlag.metadata
                 )
             }
         }
     }
 
-    private fun finnBeregningsperioder(
+    private fun beregnManueltOpphoer(
+        behandling: DetaljertBehandling,
         grunnlag: Grunnlag,
-        virkFOM: YearMonth,
-        virkTOM: YearMonth
-    ): List<Beregningsperiode> {
-        val grunnbeloep = Grunnbeloep.hentGforPeriode(virkFOM)
-        val soeskenPerioder = FinnSoeskenPeriode(grunnlag, virkFOM).hentSoeskenperioder()
-        val alleFOM = (grunnbeloep.map { it.dato } + soeskenPerioder.map { it.datoFOM } + virkTOM).map {
-            beregnFoersteFom(it, virkFOM)
-        }.distinct().sorted().zipWithNext()
-            .map { Pair(it.first, it.second.minusMonths(1)) }
-
-        val beregningsperioder = alleFOM.mapIndexed { index, (fom, tom) ->
-            val gjeldendeG = Grunnbeloep.hentGjeldendeG(fom)
-            val flokkForPeriode = hentFlokkforPeriode(fom, tom, soeskenPerioder)
-            val utbetaltBeloep = Soeskenjustering(flokkForPeriode.size, gjeldendeG.grunnbeløp).beloep
-            val søkersFødselsdato = grunnlag.soeker.hentFoedselsdato()?.verdi
-
-            val datoTom = if (index == alleFOM.lastIndex && søkersFødselsdato != null) {
-                beregnSisteTom(søkersFødselsdato, tom)
-            } else {
-                tom
-            }
-
-            Beregningsperiode(
-                delytelsesId = "BP",
-                type = Beregningstyper.GP,
-                datoFOM = fom,
-                datoTOM = datoTom,
-                grunnbelopMnd = gjeldendeG.grunnbeløpPerMåned,
-                grunnbelop = gjeldendeG.grunnbeløp,
-                soeskenFlokk = flokkForPeriode.map { it.foedselsnummer.value },
-                utbetaltBeloep = utbetaltBeloep,
-                trygdetid = 40 // TODO: Må fikses før vi tar imot saker som IKKE har 40 års trygdetid
-            )
-        }
-
-        return beregningsperioder
+        barnepensjonGrunnlag: BarnepensjonGrunnlag,
+        grunnbeloep: G
+    ): Beregning {
+        val virkningstidspunkt = virkningstidspunktFraDoedsdato(grunnlag.hentAvdoed().hentDoedsdato()?.verdi!!) // TODO
```

Comment:
Denne settes på manuelle opphør ut i fra tidligste virk vi har (så vi opphører alt), så den trengs ikke regnes ut -- man skal bare kunne bruke den man får med fra behandlingen.

Reply:
Roger that, endrer 👍